### PR TITLE
Fh 3140 add redis

### PR DESCRIPTION
**Pull request by wei-lee**
This PR will add support to save metrics data in Redis.

### CODE REVIEW

**Review by david-martin — APPROVED** [review]

File: lib/clients/influxdb.js [hunk 1]
```diff
@@ -58,12 +58,13 @@ InfluxUdp.prototype.buildMessages = function(data) {
   });
   var message = [keys.join(','), fields.join(',')];
   if (data.timestamp) {
-    message.push(data.timestamp);
+    message.push(data.timestamp*1e6); //it has to be nano seconds
```
Comment (fh-qube) [inline lib/clients/influxdb.js:61]: INFO Move this trailing comment on the previous empty line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingComment)

File: lib/clients/redis.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+/**
+ * A client that can send metrics data to a redis backend
+ * @param {Object} opts 
+ * @param {Object} opts.redisClient an instance of the redis nodejs client.
+ * @param {Number} opts.recordsToKeep the number of records to keep for each metric. Default is 1000.
+ * @param {String} opts.namespace name space for each metric key. It will be used to prefix each metric key followed by ':'. Default is `stats`. So each metric key will be like `stats:<metric key>`.
+ */
+function RedisClient(opts) {
```
Comment (fh-qube) [inline lib/clients/redis.js:15]: INFO Rename this 'RedisClient' function to match the regular expression ^[a-z][a-zA-Z0-9]*$ [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS100)

File: lib/clients/redis.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  var self = this;
+  var valueToSave = {
+    tags: message.tags,
+    fields: message.fields,
+    ts: message.timestamp
+  };
+  var redisKey = buildRedisKey(self.namespace, message.key);
+  this.client.multi([
+    ['lpush', redisKey, JSON.stringify(valueToSave)],
+    ['ltrim', redisKey, 0, self.recordsToKeep - 1],
```
Comment (fh-qube) [inline lib/clients/redis.js]: MAJOR Remove this trailing comma. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingComma)

File: lib/clients/redis.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+var BaseClient = require('./base');
+var _ = require('lodash');
+
+function buildRedisKey(namespace, metricName) {
+  return [namespace, metricName].join(':');
+}
+
+/**
+ * A client that can send metrics data to a redis backend
+ * @param {Object} opts
```
Comment (fh-qube) [inline lib/clients/redis.js]: MINOR Remove the useless trailing whitespaces at the end of this line. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingWhitespace)

**Comment by fh-qube** [comment]
SonarQube analysis reported 4 issues:
* MAJOR 1 major
* MINOR 1 minor
* INFO 2 info

Watch the comments in this conversation to review them.

**Comment by fh-qube** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* INFO 2 info

Watch the comments in this conversation to review them.

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* INFO 2 info

Watch the comments in this conversation to review them.

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* INFO 2 info

Watch the comments in this conversation to review them.

**Pull request merged**